cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -2229,6 +2229,367 @@ TEST_F(ScalarTemporalTest, TestCeilTemporal) {
   CheckScalarUnary(op, unit, times, unit, ceil_15_years, &round_to_15_years);
 }
 
+TEST_F(ScalarTemporalTest, TestCeilTemporalChangeOnBoundary) {
```

Comment:
nit, but the parameter was renamed while the tests weren't

Reply:
Done.